### PR TITLE
Changed "best modes" search algorithm

### PR DESCRIPTION
**Pull request by Diadlo**
Mentioned in #3535 by @BiTOk

### CODE REVIEW

**Comment by antis81** [comment]
LGTM

**Comment by sudden6** [comment]
Breaks mode selection for me. With the changed mode selection algorithm it doesn't find any useable resolutions for my camera (PS3 Eye).

Relevant log: https://gist.github.com/sudden6/b026c9084cdad56346b25c3279d110ec

**Comment by Diadlo** [comment]
@sudden6 Can you test it now?

**Comment by sudden6** [comment]
Retest will take till friday.

File: src/widget/form/settings/avform.cpp [hunk 1]
```diff
@@ -271,7 +271,7 @@ void AVForm::fillCameraModesComboBox()
         qDebug("width: %d, height: %d, FPS: %f, pixel format: %s\n", mode.width, mode.height, mode.FPS, pixelFormat.toStdString().c_str());
 
         if (mode.height && mode.width)
-            str += QString("%1p").arg(mode.height);
+            str += QString("%1p").arg(getModeSize(mode));
```
Comment (sudden6) [inline src/widget/form/settings/avform.cpp:274]: This is wrong, it causes the different resolutions to be shown as "4p" instead of "420p".

**Comment by sudden6** [comment]
Initial problem fixed, but new one appeared, see comment.

File: src/widget/form/settings/avform.cpp (hunk 1 above)
Comment (Diadlo) [inline src/widget/form/settings/avform.cpp:274]: Sorry. Fixed

**Pull request merged**